synfig-core/src/modules/mod_svg/svg_parser.cpp

```diff
@@ -273,180 +273,163 @@ void
+		auto build_outline_func = [&]() {
+			if (typeStroke == FILL_TYPE_NONE)
+				return;
 
-		if(typeStroke!=FILL_TYPE_NONE){//outline layer
 			if(typeStroke==FILL_TYPE_GRADIENT){
 				child_stroke=nodeStartBasicLayer(child_stroke->add_child("layer"),"stroke");
 			}
 
-			stroke_width=etl::strprintf("%f",getDimension(stroke_width)/kux);
-
-			for (const BLine& bline : k) {
-				xmlpp::Element *child_outline=child_stroke->add_child("layer");
-				child_outline->set_attribute("type","outline");
-				child_outline->set_attribute("active","true");
-				child_outline->set_attribute("version","0.3");
-				child_outline->set_attribute("desc",id);
-				build_param (child_outline->add_child("param"),"z_depth","real","0.0000000000");
-				build_param (child_outline->add_child("param"),"amount","real","1.0000000000");
-				build_param (child_outline->add_child("param"),"blend_method","integer","0");
-				build_color (child_outline->add_child("param"),getRed(stroke),getGreen(stroke),getBlue(stroke),atof(stroke_opacity.data())*atof(opacity.data()));
-				build_vector (child_outline->add_child("param"),"offset",0,0,bline.offset_id);
-				build_param (child_outline->add_child("param"),"invert","bool","false");
-				build_param (child_outline->add_child("param"),"antialias","bool","true");
-				build_param (child_outline->add_child("param"),"feather","real","0.0000000000");
-				build_param (child_outline->add_child("param"),"blurtype","integer","1");
-				//outline in nonzero
-				build_param (child_outline->add_child("param"),"winding_style","integer","0");
-
-				build_bline(child_outline->add_child("param"), bline.points, bline.loop, bline.bline_id);
-
-				build_param (child_outline->add_child("param"),"width","real",stroke_width);
-				build_param (child_outline->add_child("param"),"expand","real","0.0000000000");
-				if(stroke_linejoin.compare("miter")==0) build_param (child_outline->add_child("param"),"sharp_cusps","bool","true");
-				else build_param (child_outline->add_child("param"),"sharp_cusps","bool","false");
-				if(stroke_linecap.compare("butt")==0){
-					build_param (child_outline->add_child("param"),"round_tip[0]","bool","false");
-					build_param (child_outline->add_child("param"),"round_tip[1]","bool","false");
-				}else{
-					build_param (child_outline->add_child("param"),"round_tip[0]","bool","true");
-					build_param (child_outline->add_child("param"),"round_tip[1]","bool","true");
-				}
-				build_param (child_outline->add_child("param"),"homogeneous_width","bool","true");
-			}
+			build_outline(child_stroke, style, k, id, bline_matrix);
 
 			if(typeStroke==FILL_TYPE_GRADIENT){ //gradient in onto mode (stroke)
-				if (SVG_RESOLVE_BLINE)
-					build_fill(child_stroke,stroke,mtx);
-				else
-					build_fill(child_stroke,stroke,SVGMatrix::identity);
-			}	
+				build_fill(child_stroke, stroke, bline_matrix);
+			}
+		};
+
+		bool fill_under_stroke = true;
+		if (typeStroke != FILL_TYPE_NONE && typeStroke != FILL_TYPE_NONE) {
```

Comment:
The condition on the left and right sides is the same.

Reply:
Done